### PR TITLE
Add support for specifying underlying host interface to use for RAN network

### PR DESCRIPTION
**Pull request by ghislainbourgeois**
# Description

By default, a `macvlan` interface will use the underlying host's interface with the default gateway. When following the tutorial, all interfaces share the same underlying host's interface.

However, it prevents connectivity with the outside world for real world scenario, like connecting a real `gNodeB`, or deploying the `UPF` on the edge, away from the core.

This change allows specifying what host interface to use for each multus interfaces, allowing the operator to expose the `UPF` to external networks.

When not specifying interfaces, `multus` interfaces will be created with the `bridge` plugin, separating the network traffic in different networks.

# Checklist:

- [x] My code follows the [style guidelines](/CONTRIBUTING.md) of this project
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that validate the behaviour of the software
- [x] I validated that new and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have bumped the version of the library

### CODE REVIEW

File: src/charm.py [hunk 1]
```diff
@@ -150,6 +141,19 @@ def _on_start_simulation_action(self, event: ActionEvent) -> None:
...
+    def _network_attachment_definition_from_config(self) -> dict[str, Any]:
+        ran_nad_config = {
+            "cniVersion": "0.3.1",
+            "type": "macvlan",
+            "ipam": {"type": "static"},
+        }
+        if (gnb_interface := self._get_gnb_interface_from_config()) is not None:
+            ran_nad_config.update({"type": "macvlan", "master": gnb_interface})
+        else:
+            ran_nad_config.update({"type": "bridge", "bridge": "ran-br"})
```
Comment (saltiyazan) [inline src/charm.py:153]: Isn't it a bit confusing that the interface is `gnb_interface` but the bridge is `ran-br`
wouldn't using `gnbsim` everywhere be easier?

Just nitpicking.

**Review by saltiyazan — APPROVED** [review]

File: src/charm.py (hunk 1 above)
Comment (ghislainbourgeois) [reply src/charm.py:153]: The name of the bridge needs to be the same on the router also. We could change the name of the interface here to be `ran_interface` instead, but it might be confusing because this is really the interface towards the `UPF`.

It is already set to `ran-br` and merged in the router, but we could also create a quick PR there to change the name. Let me know what you think would be less confusing.

File: tests/unit/test_charm.py [hunk 2]
```diff
@@ -1,6 +1,7 @@
-# Copyright 2023 Canonical Ltd.
-# See LICENSE file for licensing details.
+# copyright 2023 canonical ltd.
+# see license file for licensing details.
```
Comment (gruyaume) [inline tests/unit/test_charm.py:2]: Why those changes?

File: test-requirements.txt [hunk 3]
```diff
@@ -3,7 +3,7 @@ coverage[toml]
 flake8-docstrings
 flake8-builtins
 isort
-juju==3.2.0.0
+git+https://github.com/juju/python-libjuju
```
Comment (gruyaume) [inline test-requirements.txt:6]: Why this change?

File: config.yaml [hunk 4]
```diff
@@ -3,6 +3,9 @@ options:
     type: string
     default: "192.168.251.5/24"
     description: gNodeB IP Address
+  gnb-interface:
+    type: string
+    description: Host interface to use for the RAN Network.
```
Comment (gruyaume) [inline config.yaml:8]: We many want to explicitly say what happens if this is not specified.

File: tests/unit/test_charm.py (hunk 2 above)
Comment (ghislainbourgeois) [reply tests/unit/test_charm.py]: Not sure how those changes got in there, good catch, I removed them.

File: config.yaml (hunk 4 above)
Comment (ghislainbourgeois) [reply config.yaml]: Added a quick description

File: test-requirements.txt (hunk 3 above)
Comment (ghislainbourgeois) [reply test-requirements.txt:6]: Juju 3.2.0.0 is currently broken with a pinned version of PyYAML not higher than 6.0. This fixes it because the fix is merged, but there is no release yet. I know there is work on a release happening now, I will update this PR once it is released.

Comment (ghislainbourgeois) [reply test-requirements.txt]: The fix has been merged and should be published soon, so I updated the dependency to the new version. CI will fail until it is published to PyPi.
